File: Programmers/노수일/Q12982.py
def solution(d, budget):
    d.sort()

    answer = 0
    for dest in d:
        if budget - dest < 0:
            break
        budget = budget - dest
        answer += 1

    return answer


print(solution([1, 3, 2, 5, 4], 9))
print(solution([2, 2, 3, 3], 10))

File: Programmers/정권호/Q12982.java
package programers.lv1._0512;

import java.util.Arrays;

public class Q12982 {
    class Solution {
        public int solution(int[] d, int budget) {
            int answer = 0;
            Arrays.sort(d);
            for (int i = 0; i < d.length; i++) {
                budget -= d[i];
                if (budget <= 0) {
                    if (budget == 0) {
                        i++;
                    }
                    answer = i;
                    break;
                }
            }

            if(budget > 0) {
                answer = d.length;
            }
            return answer;
        }
    }
}